src/slate/context.js

```diff
@@ -2,80 +2,11 @@
 // Licensed under the MIT license. See LICENSE file in the project root for
 // full license text.
 
-import PropTypes from 'prop-types'
 import React from 'react'
 import * as Immutable from 'immutable'
 
-/**
- * Transparent component providing current counters to its descendants in
- * a context.
- */
-export default class CounterContext extends React.Component {
-    constructor(...args) {
-        super(...args)
-
-        this.listeners = new Map()
-    }
-
-    /**
-     * Register a component for updates when counters for a node change.
-     *
-     * Note that only the last registration will be honoured.
-     *
-     * This is an internal method, you should use {@link WithCounters} instead.
-     *
-     * @param {React~Component} component
-     * @param {String} key
-     */
-    registerForCounterUpdates(component, key) {
-        this.listeners.set(component, key)
-    }
-
-    /**
-     * Unregister a component from being notified about counter changes.
-     *
-     * This is an internal method, you should use {@link WithCounters} instead.
-     *
-     * @param {React~Component} component
-     */
-    unregisterFromCounterUpdates(component) {
-        this.listeners.delete(component)
-    }
-
-    getChildContext() {
-        return {
-            counters: this.props.counters,
-            counterUpdater: this,
-        }
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.counters.equals(this.props.counters)) {
-            return
-        }
-
-        for (const [key, value] of this.props.counters.entries()) {
-            if (value.equals(prevProps.counters.get(key))) {
-                continue
-            }
-
-            for (const [listener, k] of this.listeners) {
-                if (key === k) {
-                    listener.forceUpdate()
-                }
-            }
-        }
-    }
-
-    render() {
-        return this.props.children
-    }
-}
-
-CounterContext.childContextTypes = {
-    counters: PropTypes.instanceOf(Immutable.Map),
-    counterUpdater: PropTypes.instanceOf(CounterContext),
-}
+const CountersContext = React.createContext(new Immutable.Map())
+export default CountersContext
```

Comment:
I see no reason for this name change. If you do want to change the name then I suggest `Counters`; `Counters.Provider` and `Counters.Consumer` seem much nicer than `CounterContext.Provider` and `CounterContext.Consumer`.

Reply:
But `Counters` doesn't provide any information about what is it. `NameContext` is standard for naming context